### DISCUSSION TITLE
Options to transform a few more mobs when struck by lightning

### DISCUSSION THREAD

**Discussion by Keeg-Turner** [PurpurMC/Purpur#453] [__Dismissed__] [answered]
Lightning strikes would be a good way to access some mobs that are otherwise inaccessable in survival, whilst still keeping them somewhat rare; with this in mind, I'd like to propose the following:

- An option to transform horses into zombie horses upon being struck by lightning
- An option to transform rabbits into killer bunnies upon being struck by lightning
- An option to transform wandering traders into illusioners upon being struck by lightning
- An option to transform zombies into giants upon being struck by lightning

To summarize, I'd recommend implementing the default settings like so:

```yml
world-settings:
  default:
    mobs:
      wandering_trader:
        lightning-transforms-into-illusioner: false
      zombie:
        lightning-transforms-into-giant: false
      horse:
        lightning-transforms-into-zombie-horse: false
      rabbit:
        lightning-transforms-into-killer-bunny: false
```

**Comment by Keeg-Turner** [comment]
Billy's comments on another one of my posts made me realize that this should probably be implemented in a more customizable way (i.e. have the ability to make lightning transform any mob into any other mob), but my limited coding knowledge makes me unsure of the best way to implement the options for this.

**Comment by kohy-creates** [comment]
This could be easily done with a plugin
Using EntityZapEvent and summoning a new mob

**Comment by Roan-V** [comment, marked as answer]
YouHaveTrouble just updated PurpurExtas to include this feature!
https://github.com/YouHaveTrouble/PurpurExtras/releases/tag/v1.11
